x/ethbridge/handler.go

```diff
@@ -0,0 +1,63 @@
+package ethbridge
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	"github.com/swishlabsco/cosmos-ethereum-bridge/x/ethbridge/common"
+	"github.com/swishlabsco/cosmos-ethereum-bridge/x/ethbridge/types"
+	"github.com/swishlabsco/cosmos-ethereum-bridge/x/oracle"
+)
+
+// NewHandler returns a handler for "ethbridge" type messages.
+func NewHandler(oracleKeeper oracle.Keeper, bankKeeper bank.Keeper, cdc *codec.Codec, codespace sdk.CodespaceType) sdk.Handler {
+	return func(ctx sdk.Context, msg sdk.Msg) sdk.Result {
+		switch msg := msg.(type) {
+		case MsgMakeEthBridgeClaim:
+			return handleMsgMakeEthBridgeClaim(ctx, cdc, oracleKeeper, bankKeeper, msg, codespace)
+		default:
+			errMsg := fmt.Sprintf("Unrecognized ethbridge message type: %v", msg.Type())
+			return sdk.ErrUnknownRequest(errMsg).Result()
+		}
+	}
+}
+
+// Handle a message to make a bridge claim
+func handleMsgMakeEthBridgeClaim(ctx sdk.Context, cdc *codec.Codec, oracleKeeper oracle.Keeper, bankKeeper bank.Keeper, msg MsgMakeEthBridgeClaim, codespace sdk.CodespaceType) sdk.Result {
+	if msg.CosmosReceiver.Empty() {
+		return sdk.ErrInvalidAddress(msg.CosmosReceiver.String()).Result()
+	}
+	if msg.Nonce < 0 {
+		return types.ErrInvalidEthNonce(codespace).Result()
+	}
+	if !common.IsValidEthAddress(msg.EthereumSender) {
+		return types.ErrInvalidEthAddress(codespace).Result()
+	}
+	oracleClaim := types.CreateOracleClaimFromEthClaim(cdc, types.EthBridgeClaim(msg))
+	status, err := oracleKeeper.ProcessClaim(ctx, oracleClaim)
+	if err != nil {
+		return err.Result()
+	}
+	if status.Text == oracle.SuccessStatus {
+		err = processSuccessfulClaim(ctx, bankKeeper, status.FinalClaim)
+		if err != nil {
+			return err.Result()
+		}
+	}
+	return sdk.Result{Log: status.Text}
```

Comment:
this should be returning Tags

Reply:
Do you have any documentation around Tags as a concept and their expected use? Maybe should be added to https://cosmos.network/docs/tutorial/msgs-handlers.html#msgs ?

Are you expecting something like:
```
return sdk.Result{Log: status.Text, Tags: sdk.NewTags(
			tags.Status, status,
)}

```
?

Reply:
added to https://github.com/cosmos/peggy/issues/52